### PR TITLE
[dired] add dired-git-info and hide dotfiles/details

### PR DESCRIPTION
**Pull request by mnick**
This PR makes a couple of simple changes to the dired module which I find useful:

- Add dired-git-info package and enable it automatically in git (sub)dirs
- Add support for toggling dotfiles
- Hide dotfiles and dired details by default

### CODE REVIEW

File: modules/emacs/dired/config.el [hunk 1]
```diff
@@ -168,3 +173,11 @@ we have to clean it up ourselves."
   :when (executable-find doom-projectile-fd-binary)
   :defer t
   :init (advice-add #'find-dired :override #'fd-dired))
+
+
+(use-package! dired-git-info
+  :bind (:map dired-mode-map (")" . dired-git-info-mode))
```
Comment (hlissner) [inline modules/emacs/dired/config.el]: We don't use :bind or the associated libraries in Doom. Please convert this into a `define-key` call in the `:config` block for `dired-git-info`.

```elisp
(define-key dired-mode-map ")" #'dired-git-info-mode)
```

File: modules/emacs/dired/config.el [hunk 2]
```diff
@@ -168,3 +173,11 @@ we have to clean it up ourselves."
...
+  :after dired
+  :init
+  (progn
```
Comment (hlissner) [inline modules/emacs/dired/config.el]: Please remove this superfluous `progn`.

File: modules/emacs/dired/config.el [hunk 3]
```diff
@@ -168,3 +173,11 @@ we have to clean it up ourselves."
...
+    (add-hook 'dired-after-readin-hook '+dired-enable-git-info-h)))
```
Comment (hlissner) [inline modules/emacs/dired/config.el]: On second thought, this is a mote too imposing. It is likely to break ranger/deer integration and slow TRAMP dired buffers to a crawl. I'd suggest that activating this package be opt-in, and manual.

File: modules/emacs/dired/autoload.el [hunk 4]
```diff
@@ -6,3 +6,24 @@
   (interactive)
   (mapc #'kill-buffer (doom-buffers-in-mode 'dired-mode))
   (message "Killed all dired buffers"))
+
+
+;;;###autoload
+(defun +dired-enable-git-info-h ()
+  (if (locate-dominating-file "." ".git")
```
Comment (hlissner) [inline modules/emacs/dired/autoload.el]: `locate-dominating-file` will be fairly slow in TRAMP buffers. Perhaps wrap this in a `(file-remote-p FILE)` check, and no-op in that case.

File: modules/emacs/dired/autoload.el [hunk 5]
```diff
@@ -6,3 +6,24 @@
...
+  (dired-do-kill-lines))
+
+;;;###autoload
+(defun +dired/dotfiles-toggle ()
+  (interactive)
+  (when (equal major-mode 'dired-mode)
+    (if (or (not (boundp '+dired-dotfiles-show-p)) +dired-dotfiles-show-p) ; if currently showing
+        (+dired-dotfiles-hide)
+      (progn (revert-buffer) ; otherwise just revert to re-show
+             (set (make-local-variable '+dired-dotfiles-show-p) t)))))
```
[the remark this answers is not in the document]
Comment (mnick) [reply modules/emacs/dired/autoload.el]: Good point. I removed those functions and just set `dired-omit-files` to `"^\\\."` in my config. Additionally added a localleader binding to toggle dired-omit

File: modules/emacs/dired/autoload.el (hunk 4 above)
Comment (mnick) [reply modules/emacs/dired/autoload.el]: Good catch! I added a `file-remote-p` check in the hook and enable the package only if `+ranger` is not set for the module

**Comment by mnick** [comment]
Just checking in to see if you want additional changes.

**Comment by hlissner** [comment]
Sorry for the late response. This looks good to me. Thanks again for your help!

**Pull request merged**